workers/social/lib/social/models/account.coffee

```diff
@@ -1239,3 +1257,92 @@ module.exports = class JAccount extends jraphical.Module
     options  = { limit, skip, sort }
     JSession = require './session'
     JSession.some selector, options, callback
+
+
+  destroy$: secure (client, password, callback) ->
+
+    { delegate } = client.connection
+    { profile: { nickname } } = delegate
+
+    checkUserPassword delegate, password, (err) =>
+
+      return callback new KodingError err  if err
+
+      @destroy client, callback
+
+
+  destroy: (client, callback) ->
+
+    @fetchRelativeGroups { roles: ['owner'] }, (err, groups) =>
+
+      # remove koding team
+      groups = groups.filter (group) -> group.slug isnt 'koding'
+
+      if groups.length > 1
+        return callback new KodingError 'You cannot delete your account when you have ownership in other team', groups
+
+      kallback = (label, next, err) ->
+
+        errors.push { label, err }  if err
+        next()
+
+      [ group ] = groups
+
+      errors = []
+
+      # delete the team and delete the account
+      username = @getAt 'profile.nickname'
+
+      queue = [
+
+        (next) ->
+          group.destroy client, (err) ->
+            kallback 'GroupDestroy', next, err
+
```

Comment:
we need to destroy other resources on other groups owned by this user, at this point.